qualtran/bloqs/generalized_qsp.py

```diff
@@ -12,19 +12,27 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 from functools import cached_property
-from typing import Sequence, Tuple
+from typing import Dict, Sequence, Set, Tuple, TYPE_CHECKING
 
 import cirq
 import numpy as np
-from attrs import frozen
+import scipy
+import sympy
+from attrs import field, frozen
 from numpy.polynomial import Polynomial
 from numpy.typing import NDArray
 
-from qualtran import GateWithRegisters, QBit, Register, Signature
+from qualtran import Bloq, CtrlSpec, GateWithRegisters, QBit, Register, Signature
+from qualtran.bloqs.basic_gates import Ry, ZPowGate
+from qualtran.bloqs.qubitization_walk_operator import QubitizationWalkOperator
+
+if TYPE_CHECKING:
+    from qualtran import BloqBuilder, SoquetT
+    from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
 
 
 @frozen
-class SU2RotationGate(GateWithRegisters):
+class SU2RotationGate(Bloq):
```

Comment:
Move `SU2RotationGate` to `basic_gates/` directory, override `as_cirq_op` to return a `PhasedXZGate` and override `CIRQ_GATE_TO_BLOQ_MAP` in `_cirq_gate_to_bloq` to convert a  `cirq.PhasedXZGate` to `SU2RotationGate`.  Add a builder method `.from_matrix`  so you can derive the parameters of the gate given an arbitrary 2-qubit unitary. Override `build_call_graph` and `wire_symbol` and test that `assert_decompose_is_consistent_with_t_complexity` and `bloq_autotester` works. Add a docstring (maybe just move the docstring from `rotation_matrix` method to the class) and add a `BloqExample`, update and run `dev_tools/autogenerate-bloqs-notebooks.py` to generate a doc and make sure `theta`, `phi` and `lambd` can be symbolics.